internal/controller/credentials_manager.go

```diff
@@ -60,14 +60,14 @@ func (am *CredentialsManager) Onboard(ctx context.Context, TlsSkipVerify bool, O
 	// Convert the data to JSON
 	jsonData, err := json.Marshal(data)
 	if err != nil {
-		am.Logger.Error(err, "Unable to marshal json data")
+		am.Logger.Error("Unable to marshal json data", "error", err)
```

Comment:
This doesn't do what you think. You need either to use `Errorw` (notice the `w` at the end), or explicitly pass a field:
```suggestion
		am.Logger.Errorw("Unable to marshal json data", "error", err)
```
or 
```suggestion
		am.Logger.Error("Unable to marshal json data", zap.Error(err))
```
Same with all other calls.